src/ansys/dpf/post/simulation.py

```diff
@@ -0,0 +1,524 @@
+        """
+        return self._active_selection
+
+    def activate_selection(self, selection: Selection):
+        """Sets a selection as active on the simulation.
+
+        Activating a given selection on a simulation means it is used
+        as a default selection/filter in further result queries.
+
+        Examples
+        --------
+        >>> from ansys.dpf import post
+        >>> from ansys.dpf.post import examples
+        >>> simulation = post.load_simulation(examples.static_rst)
+        >>> selection = post.selection.Selection()
+        >>> simulation.activate_selection(selection=selection)
+        >>> print(simulation.active_selection) # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+        <ansys.dpf.post.selection.Selection object at ...>
+        """
+        self._active_selection = selection
+
+    def deactivate_selection(self):
+        """Deactivate the currently active selection.
+
+        Examples
+        --------
+        >>> from ansys.dpf import post
+        >>> from ansys.dpf.post import examples
+        >>> simulation = post.load_simulation(examples.static_rst)
+        >>> selection = post.selection.Selection()
+        >>> simulation.activate_selection(selection=selection)
+        >>> print(simulation.active_selection) # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+        <ansys.dpf.post.selection.Selection object at ...>
+        >>> simulation.deactivate_selection()
+        >>> print(simulation.active_selection) # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+        None
+        """
+        self._active_selection = None
+
+    @property
+    def _time_frequencies(self):
+        """Description of the temporal/frequency analysis of the model."""
+        return self._model.metadata.time_freq_support
+
+    @property
+    def _get_time_freq_precision(self):
+        """Computes a precision for times/frequencies requests based on the underlying support."""
+        if self._time_freq_precision is None:
+            available_values = self.time_freq_support.time_frequencies.data
+            diff = np.diff(available_values, prepend=available_values[0] - 1.0)
+            minimum = np.min(diff)
+            self._time_freq_precision = minimum / 20.0
+        return self._time_freq_precision
+
+    @property
+    def time_freq_support(self):
+        """Description of the temporal/frequency analysis of the model."""
+        return self._time_frequencies
+
+    @property
+    def units(self):
+        """Returns the current time/frequency and distance units used."""
+        return self._units
+
+    def __str__(self):
+        """Get the string representation of this class."""
+        txt = (
+            "%s." % re.sub(r"(?<!^)(?=[A-Z])", " ", type(self).__name__)
+            + "\n\n\nData Sources\n------------------------------\n"
+        )
+        ds_str = self._model._data_sources.__str__()
+        txt += ds_str
+        txt += "\n\n"
+        txt += self._model.__str__()
```

Comment:
We should implement private methods in the ``self._model._data_sources`` and ``self._model`` to avoid that regex parsing.
These methods should return the partial strings.